api/observability/v1/input_types.go

```diff
@@ -0,0 +1,294 @@
+	InputTypeInfrastructure InputType = "infrastructure"
+	// InputTypeAudit contains system audit logs.
+	InputTypeAudit InputType = "audit"
+	// InputTypeReceiver defines a network receiver for receiving logs from non-cluster sources.
+	InputTypeReceiver InputType = "receiver"
+)
+
+var (
+	InputTypes = []InputType{
+		InputTypeApplication,
+		InputTypeInfrastructure,
+		InputTypeAudit,
+		InputTypeReceiver,
+	}
+)
+
+// InputSpec defines a selector of log messages for a given log type.
+type InputSpec struct {
+	// Name used to refer to the input of a `pipeline`.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Type of output sink.
+	//
+	// +kubebuilder:validation:Required
+	Type InputType `json:"type"`
+
+	// Application, named set of `application` logs that
+	// can specify a set of match criteria
+	//
+	// +kubebuilder:validation:Optional
+	// +nullable
+	Application *Application `json:"application,omitempty"`
+
+	// Infrastructure, Enables `infrastructure` logs.
+	//
+	// +kubebuilder:validation:Optional
+	Infrastructure *Infrastructure `json:"infrastructure,omitempty"`
+
+	// Audit, enables `audit` logs.
+	//
+	// +kubebuilder:validation:Optional
+	Audit *Audit `json:"audit,omitempty"`
+
+	// Receiver to receive logs from non-cluster sources.
+	//
+	// +kubebuilder:validation:Optional
+	Receiver *ReceiverSpec `json:"receiver,omitempty"`
+}
+
+type ContainerInputTuningSpec struct {
+
+	// RateLimitPerContainer is the limit applied to each container
+	// by this input. This limit is applied per collector deployment.
+	//
+	// +kubebuilder:validation:Optional
+	RateLimitPerContainer *LimitSpec `json:"rateLimitPerContainer,omitempty"`
+}
+
+// ApplicationSource defines the type of ApplicationSource log source to use.
+//
+// +kubebuilder:validation:Enum:=container
+type ApplicationSource string
+
+const (
+
+	// ApplicationSourceContainer are container logs from deployed workloads
+	// in any of the following namespaces: default, kube*, openshift*
+	ApplicationSourceContainer ApplicationSource = "container"
+)
+
+// Application log selector.
+// All conditions in the selector must be satisfied (logical AND) to select logs.
+type Application struct {
+	// Selector for logs from pods with matching labels.
+	// Only messages from pods with these labels are collected.
+	// If absent or empty, logs are collected regardless of labels.
+	//
+	// +kubebuilder:validation:Optional
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+
+	// Tuning is the container input tuning spec for this container sources
+	//
+	// +kubebuilder:validation:Optional
+	Tuning *ContainerInputTuningSpec `json:"tuning,omitempty"`
+
+	// Includes is the set of namespaces and containers to include when collecting logs.
+	// Note: infrastructure namespaces are still excluded for "*" values unless a qualifying glob pattern is specified.
+	//
+	// +kubebuilder:validation:Optional
+	Includes []NamespaceContainerSpec `json:"includes,omitempty"`
```

Comment:
Should add `listType` to this, either atomic or map, and if map, add `listMapKey` for the key field within the struct